blocks/rich-text/patterns.js

```diff
@@ -225,7 +225,8 @@ export default function( editor ) {
 		}
 
 		const block = pattern.transform( { content } );
-
-		editor.once( 'keyup', () => onReplace( [ block ] ) );
+		onReplace( [ block ] );
+		event.preventDefault();
```

Comment:
In general I'm finding `preventDefault` and `stopImmediatePropagation` to be code smells. Here their use appears legitimate, but I'd like to get in a habit of documenting their use, not unlike our "Disable reasons" for disabling ESLint rules. Here, particularly that we don't want the additional newline (`preventDefault`) and to stop TinyMCE's own processing of `keydown` which conflicts with the block replacement (since the instance is destroyed).